### PR TITLE
l2geth: allow for unprotected txs

### PR DESCRIPTION
**Pull request by tynes**
**Description**

The old transaction batch serialization assumed that the
correct chain id was used as there was a custom transaction
serialization used. Now that the full RLP of the transactions
are submitted to L1 for availability, it is possible to enable
non EIP155 transactions. This allows for universal addresses
to be used, where the same address exists across many chains.

A test is updated to test specifically that transactions that
are not protected with a chain id can be submitted and accepted
by the sequencer.

Fixes  ENG-1680

### CODE REVIEW

**Comment by tynes** [comment]
This PR will also need an update to the DTL

**Comment by tynes** [comment]
> This PR will also need an update to the DTL

Added in e84114d

File: packages/data-transport-layer/src/utils/eth-tx.ts [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
-  return ethers.BigNumber.from(v).toNumber() - 2 * chainId - 35
+  if (typeof v === 'object') {
+    v = ethers.BigNumber.from(v).toNumber()
+  }
+  // Handle unprotected transactions
+  if (v === 27 || v === 28) {
+    return v - 27
+  }
+  // Handle EIP155 transactions
+  return v - 2 * chainId - 35
```
Comment (annieke) [inline packages/data-transport-layer/src/utils/eth-tx.ts:17]: why is this transform necessary? wondering what we do to this `v` parameter that's different from i assume an l1 eth transaction

**Review by annieke — APPROVED** [review]
looks good to me!

File: packages/data-transport-layer/src/utils/eth-tx.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
   v: number | ethers.BigNumber,
   chainId: number
 ): number => {
-  return ethers.BigNumber.from(v).toNumber() - 2 * chainId - 35
+  if (typeof v === 'object') {
+    v = ethers.BigNumber.from(v).toNumber()
+  }
+  // Handle unprotected transactions
+  if (v === 27 || v === 28) {
+    return v - 27
```
Comment (smartcontracts) [inline packages/data-transport-layer/src/utils/eth-tx.ts:14]: Just need to be confident that this doesn't allow someone to play ETH transactions on L2

File: packages/data-transport-layer/src/utils/eth-tx.ts (hunk 1 above)
Comment (tynes) [reply packages/data-transport-layer/src/utils/eth-tx.ts:17]: The chainid is used to allow replay protection, see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md

File: packages/data-transport-layer/src/utils/eth-tx.ts (hunk 2 above)
Comment (tynes) [reply packages/data-transport-layer/src/utils/eth-tx.ts:14]: By not having a chainid, the transaction can be replayed. So any transactions sent to L1 without a chainid are subject to being replayed on L2, assuming the account on L2 has the correct prestate (nonce/value to cover gas)

**Comment by tynes** [comment]
e579cf2 adds test coverage for replicas, it enables the replica in the integration tests which adds a bit of additional load to the machine that has to run the integration tests

**Review by smartcontracts — DISMISSED** [review]
LGTM

**Comment by smartcontracts** [comment]
Actually there are failing integration tests that need to be solved first

**Review by smartcontracts — CHANGES_REQUESTED** [review]
Blocking merge of this PR pending additional tests, will have to wait until after upgrade

**Comment by tynes** [comment]
CI is failing due to an http client not being able to connect properly to the replica. It is unclear why this is happening because the tests work without a problem on my local machine. Is there a max number of ports that can be listening when running on Github actions? Perhaps there is something that we are missing with the docker compose config.

```
 96 passing (10m)
  1 pending
  2 failing

  1) Replica Tests
       Matching blocks
         should sync a transaction:
     Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.4.4)
      at Logger.makeError (/optimism/node_modules/@ethersproject/logger/src.ts/index.ts:213:28)
      at Logger.throwError (/optimism/node_modules/@ethersproject/logger/src.ts/index.ts:225:20)
      at JsonRpcProvider.<anonymous> (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:396:23)
      at step (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:48:23)
      at Object.throw (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:29:53)
      at rejected (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:21:65)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  2) Replica Tests
       Matching blocks
         sync an unprotected tx (eip155):
     Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.4.4)
      at Logger.makeError (/optimism/node_modules/@ethersproject/logger/src.ts/index.ts:213:28)
      at Logger.throwError (/optimism/node_modules/@ethersproject/logger/src.ts/index.ts:225:20)
      at JsonRpcProvider.<anonymous> (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:396:23)
      at step (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:48:23)
      at Object.throw (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:29:53)
      at rejected (/optimism/node_modules/ethers/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:21:65)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

```

**Review by smartcontracts — APPROVED** [review]

**Comment by hellwolf** [comment]
Great!

how could we utilize this to deploy the 1820 contract? https://kovan-optimistic.etherscan.io/address/0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24

**Pull request merged**